src/Security/Trust.cs

```diff
@@ -128,17 +128,43 @@ void Initialize (IntPtr certHandle, SecPolicy policy)
 				throw new ArgumentException (result.ToString ());
 		}
 
-		[Deprecated (PlatformName.iOS, 12,1)]
-		[Deprecated (PlatformName.TvOS, 12,1)]
-		[Deprecated (PlatformName.WatchOS, 5,1)]
-		[Deprecated (PlatformName.MacOSX, 10,14,1)]
+#if !NET
+		[Deprecated (PlatformName.iOS, 12,1, message: "Use 'SecTrustEvaluateWithError' instead.")]
+		[Deprecated (PlatformName.TvOS, 12,1, message: "Use 'SecTrustEvaluateWithError' instead.")]
+		[Deprecated (PlatformName.WatchOS, 5,1, message: "Use 'SecTrustEvaluateWithError' instead.")]
+		[Deprecated (PlatformName.MacOSX, 10,14,1, message: "Use 'SecTrustEvaluateWithError' instead.")]
+#else
+		[UnsupportedOSPlatform ("ios12.1")]
+		[UnsupportedOSPlatform ("tvos12.1")]
+		[UnsupportedOSPlatform ("macos10.14.1")]
+#if IOS
+		[Obsolete ("Starting with ios12.1 Use 'SecTrustEvaluateWithError' instead.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

Comment:
The managed method is `Evaluate (out NSError)`:

https://github.com/xamarin/xamarin-macios/blob/e0cccca2f16d563447c20373d910987dc93c3068/src/Security/SecTrust.cs#L301

so that's what the message should say (and not `SecTrustEvaluateWithError`).